src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -3828,7 +3825,7 @@ internal static int LastIndexOfAnyExceptInRangeUnsignedNumber<T>(ref T searchSpa
                     ref T oneVectorAwayFromEnd = ref Unsafe.Subtract(ref end, Vector128<T>.Count);
                     do
                     {
-                        count += BitOperations.PopCount(Vector128.Equals(Vector128.LoadUnsafe(ref current), targetVector).ExtractMostSignificantBits());
+                        count += Vector128.Equals(Vector128.LoadUnsafe(ref current), targetVector).GetMatchCount();
```

Comment:
This should especially help ARM64. Previously, this was doing movemask emulation only to load the scalar value back into a SIMD register to do popcount.